modules/core/src/main/java/org/apache/synapse/mediators/MediatorWorker.java

```diff
@@ -86,7 +97,25 @@ public void run() {
                 debugManager.advertiseMediationFlowStartPoint(synCtx);
             }
 
-            seq.mediate(synCtx);
+            // If this is a scatter message, then we need to use the continuation state and continue the mediation
```

Comment:
can we evaluate boolean result = seq.mediate(synCtx); without any check and modify the check to if (isScatterMessage(synCtx) && result ) {

Reply:
Fixed with 1b4379f